### DISCUSSION TITLE
If you could add one thing to Appwrite's docs, what would it be?

### DISCUSSION THREAD

**Discussion by gewenyu99** [appwrite/appwrite#5692] [Ideas]
Hi all!

We're working on the next generation of Appwrite's docs to address some of the pain points and limitations we currently face.

This is your chance to voice your wildest wishes about Appwrite's docs. You can ask for one (1) item, so make it count!

Please avoid duplicating wishes that others have already made, but feel free to upvote them!

We'll be taking these wishes into account when we implement our update.

Cheers,

Vincent

**Comment by eldadfux** [comment]
Search.

**Reply by gewenyu99** [reply]
s a m e

**Comment by safwanyp** [comment]
A better layout for the different endpoints in an API.

For example, if I go to the Database API server-side docs and click on something that's really close to the end on the page, I have to scroll all the way back up to look for the other endpoints.

Not sure what could be a better layout though 😅

**Reply by eldadfux** [reply]
Let the Appwrite design team deal with that, this should be an interesting challenge :)

**Reply by safwanyp** [reply]
> Let the Appwrite design team deal with that, this should be an interesting challenge :)

I agree 😜 Genuinely though, I feel like a simple "scroll to top" button would be perfect for this. This will help keep the current UX as well!

**Reply by gewenyu99** [reply]
Who said your wild wishes need to be practical LOL

**Reply by eldadfux** [reply]
We actually do have this, but I still think it can be better.
<img width="1438" alt="image" src="https://github.com/appwrite/appwrite/assets/1297371/5db7c5bb-0e3f-41d1-ae8d-473cf4a9228c">

**Reply by gewenyu99** [reply]
I've never used that button or noticed it XD

**Reply by ashuvssut** [reply]
where is the current docs website repo?

**Reply by Suven-p** [reply]
![image](https://github.com/appwrite/appwrite/assets/47393058/2b4b8a2f-a247-4f81-a750-c00e17503837)
Navigation bar with fixed position?

**Reply by Suven-p** [reply]
![image](https://github.com/appwrite/appwrite/assets/47393058/3abd1a7d-02ba-41b4-a77a-9e377be9f429)
This might have been a better example

**Comment by byawitz** [comment]
References & APIs list.
A mega-links page with all links to, functions, models, variables, and endpoints.

For example _Just similar_
[Vue](https://vuejs.org/api/)
[React](https://react.dev/reference/react)
[Firebase](https://firebase.google.com/docs/reference)
[Flutter](https://api.flutter.dev/)

**Reply by eldadfux** [reply]
Pretty cool, I liked the Vue one the most!

**Comment by Susmita-Dey** [comment]
I tried to redesign the Appwrite docs earlier but due to college stuffs couldn't make it. But here's what I came out at that point during the research phase. Hope this will help you to make Appwrite docs better in the next version.


Link to Figma file(representing pain points):- https://www.figma.com/file/rjgUeRP3fn8YqhKE8h0FN2/Appwrite-Docs-Redesign?type=design&node-id=10%3A6060&t=pSyPjhMRyvUQ8Y0C-1

Link to redesign attempt file:- https://www.figma.com/file/rjgUeRP3fn8YqhKE8h0FN2/Appwrite-Docs-Redesign?type=design&node-id=0%3A1&t=pSyPjhMRyvUQ8Y0C-1

Also, in addition to the above, I'd like to add live working examples for each API or functions(if possible)

**Reply by gewenyu99** [reply]
O wow that's amazing. Maybe you can try working on this during Hacktoberfest this year, if we participate again. We always wanted to include designers in the OSS world!

**Reply by gewenyu99** [reply]
The live examples thing is definitely something I want to see, too!

**Reply by Susmita-Dey** [reply]
Thank you @gewenyu99 Hope these ideas would help to make better docs. Meanwhile I'm a developer by passion and designer by hobby. 😄

**Comment by sergioponguta** [comment]
Playground on every endpoint, probably you can create a base app, hidden of public, but that you can edit and hit to see the expected results. 

Probably a lot of work due to the amount of SDKs.

**Reply by gewenyu99** [reply]
👍 I personally want to do this with Web at the very least, if we get time :P

I believe this will be mostly good enough. The behavior across our SDKs are very consistent.

Playground for Apple, for example, might be much harder.

Do you know of some other services that have this in docs? I want to peep to see what/how they do this.

**Comment by gepd** [comment]
A recipes section with examples like, authentication, pagination, search, upload, validation with functions, etc

**Reply by gewenyu99** [reply]
How do you feel about something with a similar purpose, like a implementation steps section for each service.

It would walk through the steps you take to add a service to your app.

I think this kind of high level information is 100% direly missing from our docs.

**Reply by gepd** [reply]
Yeah is kind of that, the idea is to include things like we always have to do, for example, in pagination we can use offset and cursor. Currently, we have both examples, but they're not complete as isn't the intention of the documentation at the moment. The receipt should cover things like infinite pagination, previous, next buttons, etc. In other words completelly functional code

**Reply by gewenyu99** [reply]
> Yeah is kind of that, the idea is to include things like we always have to do, for example, in pagination we can use offset and cursor. Currently, we have both examples, but they're not complete as isn't the intention of the documentation at the moment. The receipt should cover things like infinite pagination, previous, next buttons, etc. In other words completelly functional code

Do you have examples of how others do this? I can take inspiration ;)

**Reply by gepd** [reply]
Sure, I like this two examples:

https://mantine.dev/form/recipes/ 
https://xstate.js.org/docs/recipes/react.html

I like the mantine docs, specially that code is in the center, it make feel like it's important and I have to focus on it, in appwrite now I'm used to docs, but at the beginning the experience wasn't so good because the code is at the right. I'd move the code after the params description

**Reply by gewenyu99** [reply]
Thank you! I'll take a look <3

**Comment by ashuvssut** [comment]
I know the website is probably not gonna get Implemented it in Next.js / React. But, I would like to do a lot of contributions if implemented in Next.js / React

**Reply by gewenyu99** [reply]
It's probably going to be Markdown with JS/HTML mixins, and can be built easily for contributors.

That's one of the big issues we're trying to solve, is maintainability, especially for people outside our team <3

**Reply by Suven-p** [reply]
Does this involve static site generators?

**Reply by ashuvssut** [reply]
Check out this personal blog page of Josh Comeau https://www.joshwcomeau.com/react/why-react-re-renders/

It's written in [Mdx.js](https://mdxjs.com/) (MarkDown with TypeScript JSX) with React. You can add JSX component right into Markdown. Its an awesome library to create interactable Markdown Websites

**Reply by gewenyu99** [reply]
> Check out this personal blog page of Josh Comeau https://www.joshwcomeau.com/react/why-react-re-renders/
> 
> It's written in [Mdx.js](https://mdxjs.com/) (MarkDown with TypeScript JSX) with React. You can add JSX component right into Markdown. Its an awesome library to create interactable Markdown Websites

We might do MDX, or something similar. Probably not react, since console is written in svelte, we have no reason to tack on a new framework.

Anyway, we'll figure something out :P

**Comment by Suven-p** [comment]
A copy-pastable example for REST api using `curl`, `wget` and/or its equivalent in powershell

**Reply by ashuvssut** [reply]
Fork this Postman collection:- https://www.postman.com/interstellar-water-360005/workspace/aaaa/collection/5190147-6ec0cc3a-a363-42bd-9b02-cffc6164fa70?action=share&creator=14750484

Not only `curl` / `wget` but also you can get code generated in any language by postman to help you make requests

**Reply by ashuvssut** [reply]
note that when sending requests, sending session cookie works fine. You do not necessarily need to request for jwt before requesting any endpoint. That Postman collection works great. You just need to update the api endpoint variable to the new appwrite api enpoint url

**Reply by gewenyu99** [reply]
We almost did this, actually. But it's almost impossible to do chunked uploads the way we implmented them with curl.

In fact I think we generated cURL examples in our SDK generators but never exposed them 😆 We can try to look into this, but keep in mind the experience is going to be terrible with session management and all :P

**Reply by Suven-p** [reply]
Ahh. I think the playground idea above makes more sense.

**Reply by fliitor** [reply]
You can got it from https://github.com/appwrite/appwrite/tree/master/app/config/specs
+1 for live swagger playground

**Reply by gewenyu99** [reply]
Play ground would be nice. Maybe shell projects with prepopulated data you can clone to follow tutorials with, too?

**Comment by fliitor** [comment]
1. An update 30 days with appwrite website
2. A "how to" tutorial with common github/discord questions
3. Link from doc to github source code
4. A "create your own runtime" section
5. An "Architecture" part, based on [Appwrite’s building blocks](https://dev.to/appwrite/30daysofappwrite-appwrite-s-building-blocks-1936)

**Reply by gewenyu99** [reply]
If you have to pick one from this list, what would it be.

**Reply by fliitor** [reply]
4. for sure. Probably the most complicated... Facilitating the creation of runetime would open up appwrite to new horizons.

**Reply by gewenyu99** [reply]
This already exists ;) we'd just package this into our docs. https://github.com/open-runtimes/open-runtimes/blob/main/docs/add-runtime.md

What's your next item.

**Reply by fliitor** [reply]
Know cause we use it to create our own runtime, but it's not very beginner friendly :)
Example with override current runtime would be interesting
Or any other documentation that gives you more runtimes, and not just from the appwrite team.

**Reply by gewenyu99** [reply]
It probably will not be available for Cloud, just because it would be our infrastructure, we'll see what we can do for self-hosting.

Creating your own runtime in general will be tricky to do :P

Thanks for the suggestion!

**Comment by raman04-byte** [comment]
Spoon feeding to the new users by helping them how they can create themselves.

**Reply by gewenyu99** [reply]
New appwrite docs will have chocolate and vanilla rice pudding. I'll try to spoon feed as many as I can!
